core/server/web/api/app.js

```diff
@@ -14,16 +15,27 @@ module.exports = function setupApiApp() {
     }
 
     // If there is a version in the URL, and this is a valid API URL containing admin/content
-    // Then 307 redirect (preserves the HTTP method) to a versionless URL with `accept-version` set.
-    apiApp.all('/:version(v2|v3|v4|canary)/:api(admin|content)/*', (req, res) => {
-        const {version} = req.params;
-        const versionlessURL = req.originalUrl.replace(`${version}/`, '');
-        if (version.startsWith('v')) {
-            res.header('accept-version', `${version}.0`);
-        } else {
-            res.header('accept-version', version);
+    // Rewrite the URL and add the accept-version & deprecated headers
+    apiApp.all('/:version(v2|v3|v4|canary)/:api(admin|content)/*', (req, res, next) => {
+        let {version} = req.params;
+        const versionlessUrl = req.url.replace(`${version}/`, '');
+
+        // Always send the explicit, numeric version in headers
+        if (version === 'canary') {
+            version = 'v4';
         }
-        res.redirect(307, versionlessURL);
+
+        // Rewrite the url
+        req.url = versionlessUrl;
+
+        // Add the accept-version header so our internal systems will act as if it was set on the request
+        req.headers['accept-version'] = `${version}.0`;
+
+        res.header('Deprecation', `version="${version}"`);
```

Comment:
Would be super cool to add a link to our docs here like the [Deprecation header spec](https://tools.ietf.org/id/draft-dalal-deprecation-header-03.html#documentation) for `Link` suggests. This might be more useful in the 4.x branch but also while looking at the Deprecation header spec setting `Sunset` would be awesome too. 

All of this is nice to have though. Might make someones migration experience slightly nicer :)

Reply:
Oh and read the above as, "what are your thoughts on these additional headers" rather than "it should be done this way, period" :smile:

Reply:
Yeah I was thinking we'd add that in once we do have docs. We don't have anything to link to right now. Have added a note to https://github.com/TryGhost/Toolbox/issues/307